### ISSUE TITLE
Proposal: explicit JSON parameter typing for Aurora PostgreSQL Data API

### ISSUE THREAD

**Issue by mikelamutxastegi** [krotscheck/go-rds-driver#13]
## Problem

The driver currently maps Go `string` values to Data API `StringValue` parameters.

For Aurora PostgreSQL, this can be insufficient when a parameter represents JSON/JSONB. The Data API supports `TypeHint: JSON`, but the driver cannot always infer that an arbitrary Go string should be sent as JSON.

This differs from PostgreSQL drivers such as pgx. With the PostgreSQL wire protocol, the server resolves a placeholder type from SQL context and the target column. Given:

```sql
INSERT INTO example (metadata, description) VALUES ($1, $2)
```

where `metadata` is `jsonb` and `description` is `text`, PostgreSQL can resolve `$1` as `jsonb` and `$2` as `text` even when both Go values are identical strings.

The Data API driver constructs `SqlParameter` values directly, so it does not receive this schema-derived parameter type information.

## Why automatic JSON detection is risky

Detecting JSON by inspecting every string can produce false positives:

```go
metadata := `{"mode":"strict"}`    // intended for a JSONB column
description := `{"mode":"strict"}` // intended as ordinary TEXT
```

Both values are valid JSON, but their intended database types differ.

## Proposal

Consider supporting an explicit JSON value type, `json.RawMessage`, or both. The driver would map the opt-in value to:

```go
types.SqlParameter{
    TypeHint: types.TypeHintJson,
    Value: &types.FieldMemberStringValue{
        Value: string(jsonValue),
    },
}
```

For example, a driver-specific API could be:

```go
type JSON string

func NewJSON(value string) (JSON, error) {
    if !json.Valid([]byte(value)) {
        return "", errors.New("invalid JSON")
    }
    return JSON(value), nil
}
```

Usage:

```go
metadata, err := rds.NewJSON(`{"mode":"strict"}`)
if err != nil {
    return err
}

_, err = db.ExecContext(
    ctx,
    `UPDATE example SET metadata = $1 WHERE id = $2`,
    metadata,
    id,
)
```

The implementation may need `driver.NamedValueChecker` so the explicit marker survives `database/sql` conversion before the driver builds the Data API parameter.

## ORM compatibility

An ORM can preserve intent through a custom serializer that returns `json.RawMessage` only for fields declared as JSON/JSONB. This remains compatible with standard PostgreSQL drivers: PostgreSQL continues to infer `jsonb` versus `text` from the target column, while this driver maps the explicit marker to `TypeHintJson`.

This avoids globally treating JSON-looking strings as JSON parameters.

## Open questions

1. Is explicit JSON typing desirable as part of the public API?
2. Should the driver support a driver-specific `JSON` type, `json.RawMessage`, or both?
3. Should JSON support be limited to explicit opt-in rather than automatic detection?
4. Are there compatibility concerns for Aurora PostgreSQL and Aurora MySQL?

I would be happy to contribute an implementation and focused tests once the preferred API direction is clear.
